front/src/modules/pathfinding/utils.ts

```diff
@@ -176,3 +176,21 @@ export const isStation = (chCode: string): boolean =>
   chCode === 'BV' || chCode === '00' || chCode === '';
 
 export const isPathStepInvalid = (step: PathStep | null): boolean => step?.isInvalid || false;
+
+export function mergeDuplicatedOperationalPoints<T>(
+  operationalPoints: T[],
+  getDuplicatedOP: (operationalPoint: T) => string,
+  mergeOP: (previousOperationalPoint: T, nextOperationalPoint: T) => void
+): T[] {
+  return Object.values(
+    operationalPoints.reduce<Record<string, T>>((mergedOperationalPoints, operationalPoint) => {
```

Comment:
Maybe it's more understandable

```suggestion
    operationalPoints.reduce<Record<string, T>>((uniqueOperationalPoints, operationalPoint) => {
```

Reply:
done